src/Chapter1.hs

```diff
@@ -553,8 +554,17 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+tripleApply :: (Int -> Int -> Int) -> Int -> Int -> Int -> Int
```

Comment:
I just guessed the syntax here based on the type declaration and rules specified above. The mid implementation still feels too complicated (and has too much repetition). I'm guessing we'll learn to optimise it upcoming chapters.